include/alpaka/pltf/PltfOmp5.hpp

```diff
@@ -0,0 +1,105 @@
+ */
+
+#pragma once
+
+#ifdef ALPAKA_ACC_ANY_BT_OMP5_ENABLED
+
+#if _OPENMP < 201307
+    #error If ALPAKA_ACC_ANY_BT_OMP5_ENABLED is set, the compiler has to support OpenMP 4.0 or higher!
+#endif
+
+#include <alpaka/pltf/Traits.hpp>
+#include <alpaka/dev/DevOmp5.hpp>
+#include <alpaka/core/Concepts.hpp>
+
+#include <sstream>
+#include <vector>
+#include <limits>
+
+namespace alpaka
+{
+    namespace pltf
+    {
+        //#############################################################################
+        //! The CPU device platform.
+        class PltfOmp5 :
+            public concepts::Implements<ConceptPltf, PltfOmp5>
+        {
+        public:
+            //-----------------------------------------------------------------------------
+            ALPAKA_FN_HOST PltfOmp5() = delete;
+        };
+    }
+
+    namespace dev
+    {
+        namespace traits
+        {
+            //#############################################################################
+            //! The CPU device device type trait specialization.
+            template<>
+            struct DevType<
+                pltf::PltfOmp5>
+            {
+                using type = dev::DevOmp5;
+            };
+        }
+    }
+    namespace pltf
+    {
+        namespace traits
+        {
+            //#############################################################################
+            //! The CPU platform device count get trait specialization.
+            template<>
+            struct GetDevCount<
+                pltf::PltfOmp5>
+            {
+                //-----------------------------------------------------------------------------
+                ALPAKA_FN_HOST static auto getDevCount()
+                -> std::size_t
+                {
+                    ALPAKA_DEBUG_FULL_LOG_SCOPE;
+
+                    const std::size_t count = static_cast<std::size_t>(::omp_get_num_devices());
+                    // runtime will report zero devices if the host is the target (or without offloading)
+                    return count > 0 ? count : 1;
+                }
+            };
+
+            //#############################################################################
+            //! The CPU platform device get trait specialization.
+            template<>
+            struct GetDevByIdx<
+                pltf::PltfOmp5>
+            {
+                //-----------------------------------------------------------------------------
+                //! \param devIdx device id, less than GetDevCount, will be set to omp_get_initial_device() otherwise
+                ALPAKA_FN_HOST static auto getDevByIdx(
+                    std::size_t devIdx)
+                -> dev::DevOmp5
+                {
+                    ALPAKA_DEBUG_FULL_LOG_SCOPE;
+
+                    std::size_t const devCount(static_cast<std::size_t>(::omp_get_num_devices()));
+                    int devIdxOmp5 = static_cast<int>(devIdx);
```

Comment:
What is the mapping between alpaka `devIdx` and OpenMP? Is `devIdx == 0` always the host and `devIdx > 0` the offloading devices?
Can we document this?

Reply:
https://github.com/alpaka-group/alpaka/commit/2db34b82d4586a5ea853005aba532e9b5ddfccf2#diff-0af16922c3afe4ac9df8b1a1c77c3229R84

Changed the docstring a bit.